cpp/ql/src/semmle/code/cpp/dataflow/internal/FlowVar.qll

```diff
@@ -138,18 +138,42 @@ private module PartialDefinitions {
   }
 
   /**
-   * A partial definition that's a definition by reference (in the sense of the
-   * `definitionByReference` predicate).
+   * A partial definition that's a definition by reference.
    */
   class DefinitionByReference extends PartialDefinition, TReferenceArgument {
     VariableAccess va;
 
-    DefinitionByReference() { definitionByReference(va, definedExpr) }
+    DefinitionByReference() { referenceArgument(va, definedExpr) }
```

Comment:
It might look strange that `this` is not restricted in this charpred. That's because the full extent of this class includes the charpred of the superclass, which relates `this` to `definedExpr`, and `va` is functionally determined by `definedExpr`.

Reply:
I've added a commit that turns the above remark into a code comment.